scripts/west_commands/build.py

```diff
@@ -99,8 +99,6 @@ def do_add_parser(self, parser_adder):
         # flags
 
         parser.add_argument('-b', '--board', help='board to build for')
-        # Hidden option for backwards compatibility
-        parser.add_argument('-s', '--source-dir', help=argparse.SUPPRESS)
```

Comment:
I totally get what you're doing here, and although I use this option every day, I would be willing to part ways with it in the name of making progress on this more important issue (assuming "snippet" is chosen).

But there are a few places in the docs that are using still:

```
doc/guides/west/sign.rst
29:   west build -b YOUR_BOARD -s bootloader/mcuboot/boot/zephyr -d build-mcuboot
30:   west build -b YOUR_BOARD -s zephyr/samples/hello_world -d build-hello-signed -- \

doc/guides/dts/howtos.rst
35:   west build -b qemu_cortex_m3 -s samples/hello_world --cmake-only
```

Those would need cleanup as well, pending resolution of the bikeshed debate.